stdlib/public/runtime/ImageInspection.h

```diff
@@ -106,6 +109,37 @@ void addImageAccessibleFunctionsBlockCallbackUnsafe(const void *baseAddress,
 
 int lookupSymbol(const void *address, SymbolInfo *info);
 
+#if defined(_WIN32)
+/// Configure the environment to allow calling into the Debug Help library.
+///
+/// \param body A function to invoke. This function attempts to first initialize
+///   the Debug Help library. The result of that operation is passed to this
+///   function.
+///
+/// \returns Whether or not the Debug Help library was successfully initialized
+///   and \a body was subsequently invoked.
+///
+/// On Windows, the Debug Help library (DbgHelp.lib) is not thread-safe. All
+/// calls into it from the Swift runtime and stdlib should route through this
+/// function.
+SWIFT_RUNTIME_STDLIB_SPI
+bool _swift_withDebugHelpLibrary(
+  const std::function<void(bool /* isInitialized */)>& body);
+
+/// As with \c _swift_withDebugHelpLibrary(), but returning a value.
+template <typename F>
+static inline auto
+withDebugHelpLibrary(const F& body) -> decltype(body(false)) {
```

Comment:
Convenience wrapper around `_swift_withDebugHelpLibrary()` that returns some value.